### PR TITLE
Including more general DQM compare modules for pixel tracks objects

### PR DESCRIPTION
**Pull request by borzari**
#### PR description:

This PR is the replacement of https://github.com/cms-sw/cmssw/pull/43964 with essentially the same changes, **except** for the Alpaka vs CUDA compare modules that are not needed anymore. It still includes generalized DQM compare modules for pixel rechits, tracks and vertices, which are templated if needed for other SoA types, an HLT customize function to include the module changes and pixel digi errors monitoring (that is also included in the DQM step), and the changes requested in https://github.com/cms-sw/cmssw/pull/43964#discussion_r1491530401.

#### PR validation:

Executed the code with wf `12834.403` to check DQM plots and everything works as expected.

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

This is not a backport. These changes will be backported to 14_0_X in a following PR.

Pinging @mmusich @makortel @fwyzard @AdrianoDee

### CODE REVIEW

File: HLTrigger/Configuration/python/customizeHLTforCMSSW.py [hunk 1]
```diff
@@ -260,6 +260,90 @@ def checkHLTfor43774(process):
                     print('# TSG WARNING: check value of parameter "useAbs" in',filt,'(expect True but is False)!')
 
     return process
+
+def customizeHLTforPR(process):
```
Comment (mmusich) [inline HLTrigger/Configuration/python/customizeHLTforCMSSW.py:264]:
```suggestion
def customizeHLTfor45206(process):
```

File: HLTrigger/Configuration/python/customizeHLTforCMSSW.py [hunk 2]
```diff
@@ -326,5 +410,6 @@ def customizeHLTforCMSSW(process, menuType="GRun"):
     process = checkHLTfor43774(process)
     process = customizeHLTfor44576(process)
     process = customizeHLTfor45063(process)
+    process = customizeHLTforPR(process)
```
Comment (mmusich) [inline HLTrigger/Configuration/python/customizeHLTforCMSSW.py:413]:
```suggestion
    process = customizeHLTfor45206(process)
```

File: DQM/SiPixelHeterogeneous/plugins/SiPixelCompareRecHitsSoAAlpaka.cc [hunk 3]
```diff
@@ -1,244 +0,0 @@
-#include "DQMServices/Core/interface/MonitorElement.h"
```
Comment (mmusich) [inline DQM/SiPixelHeterogeneous/plugins/SiPixelCompareRecHitsSoAAlpaka.cc:1]: @borzari, please do NOT delete the old plugins.
If you do that, when we we'll run (from the TSG side) the confDB parsing in next pre-release, the modules currently in the HLT menu will disappear, leading to have to redo the sequence by hand with the new plugins.
In the past what we did was:
   * add the new plugins in the master (and backport down to the cycle used by HLT - currently CMSSW_14_0_X)    
   * JIRA ticket to change the sequence in ConfDB
   * PR to `cmssw` to remove the old plugins in the folllowing pre-release (that could be done by TSG).
   
See e.g. https://github.com/cms-sw/cmssw/pull/40646 for some past instance.

Comment (borzari) [reply DQM/SiPixelHeterogeneous/plugins/SiPixelCompareRecHitsSoAAlpaka.cc]: @mmusich I already reverted to the previous commit when the files weren't deleted and made the changes requested in your comments above, regarding changing the name of function `customizeHLTforPR` to `customizeHLTfor45206`. I am also going to clarify in the description of the PR that this will be backported to 14_0_X. Thank you for the suggestions

File: DataFormats/TrackSoA/src/classes_def.xml [hunk 4]
```diff
@@ -5,9 +5,7 @@
   <class name="reco::TrackSoAView<pixelTopology::Phase1>"/>
 
   <class name="PortableHostCollection<reco::TrackLayout<pixelTopology::Phase1>>"/>
-  <class name="pixelTrack::TracksHostPhase1" ClassVersion="3">
-    <version ClassVersion="3" checksum="794224446"/>
-  </class>
+  <class name="pixelTrack::TracksHostPhase1"/>
```
Comment (mmusich) [inline DataFormats/TrackSoA/src/classes_def.xml:8]: out of curiosity (even if this lies beyond the HLT proper review) why these changes to the class version are needed?

Comment (borzari) [reply DataFormats/TrackSoA/src/classes_def.xml:8]: I followed this comment https://github.com/cms-sw/cmssw/pull/43964#discussion_r1491530401 from the previous PR. Maybe @makortel can provide more specific information about why this is the case now

**Comment by mmusich** [comment]
test parameters:
   * workflow_opts_gpu= -w upgrade
   * workflows_gpu= 12434.403, 12434.405, 12434.503

**Comment by mmusich** [comment]
@cmsbuild, please test

File: DataFormats/TrackSoA/src/classes_def.xml (hunk 4 above)
Comment (fwyzard) [reply DataFormats/TrackSoA/src/classes_def.xml:8]: Quoting @makortel:
> Given that `TrackingRecHitHostPhase1` itself is an instantiation of a class template, its version should not be tracked this way (but with `CMS_CLASS_VERSION()` in the class template definition). But given that schema evolution of `PortableHostCollection` is unexplored anyway (and therefore nothing is promised), I'd suggest to remove the class version for now (until the schema evolution is properly understood)

**Comment by mmusich** [comment]
enable gpu

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
enable gpu

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
> -1

damn, can't catch a break...

**Comment by mmusich** [comment]
test parameters:
   * workflow_opts_gpu= -w upgrade
   * workflows_gpu= 12434.403, 12434.503, 12634.403

**Comment by borzari** [comment]
@cmsbuild, please test

**Comment by borzari** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
> DQMHistoSizes: changed ( 12434.403,... ): 55243.484 KiB SiPixelHeterogeneous/PixelErrorCompareGPUvsCPU
> DQMHistoSizes: changed ( 12434.403,... ): 37.574 KiB SiPixelHeterogeneous/PixelErrorsCPU
> DQMHistoSizes: changed ( 12434.403,... ): 37.574 KiB SiPixelHeterogeneous/PixelErrorsGPU
> DQMHistoSizes: changed ( 12434.403,... ): 0.020 KiB SiPixelHeterogeneous/PixelTrackCompareDeviceVSHost

the first three changes are due to new additions to the relvals (these plots were not filled before, thanks for adding them!) while the fourth seems likely to come to some tiny residual discrepancy in the gpu vs cpu tracking code.

**Comment by mmusich** [comment]
+hlt
   * see https://github.com/cms-sw/cmssw/pull/45206#issuecomment-2179104733

**Comment by jfernan2** [comment]
+1

**Comment by fwyzard** [comment]
+heterogeneous

**Comment by mmusich** [comment]
@cms-sw/dqm-l2 please review this PR, it would be useful to have in the next 14.1.X pre-release. 
@borzari can you please open a backport to CMSSW_14_0_X? If there are no objections, I think we should try to integrate this in the next 2024 menu (V1.4) @cms-sw/hlt-l2, @missirol FYI.

**Comment by Martin-Grunewald** [comment]
Once the PR(s) are integrated, I assume an update of https://its.cern.ch/jira/browse/CMSHLT-3147 is warranted, to get the updated path configuration.

**Comment by mmusich** [comment]
> I assume an update of https://its.cern.ch/jira/browse/CMSHLT-3147 is warranted, to get the updated path configuration.

I will have to be a different ticket because since the persistence issue (see https://github.com/cms-sw/cmssw/issues/44700) is not solved we still can't kick these plugins out of the menu. They will be simply replaced by this (better) version.
I can take care of opening the ticket.

**Comment by tjavaid** [comment]
+1

**Comment by antoniovilela** [comment]
+1
